llvm/include/llvm/IR/IntrinsicsDirectX.td

```diff
@@ -83,6 +83,7 @@ def int_dx_umad : DefaultAttrsIntrinsic<[llvm_anyint_ty], [LLVMMatchType<0>, LLV
 def int_dx_normalize : DefaultAttrsIntrinsic<[LLVMMatchType<0>], [llvm_anyfloat_ty], [IntrNoMem]>;
 def int_dx_rsqrt  : DefaultAttrsIntrinsic<[llvm_anyfloat_ty], [LLVMMatchType<0>], [IntrNoMem]>;
 def int_dx_wave_is_first_lane : DefaultAttrsIntrinsic<[llvm_i1_ty], [], [IntrConvergent]>;
+def int_dx_wave_read_lane_at : DefaultAttrsIntrinsic<[llvm_any_ty], [LLVMMatchType<0>, llvm_i32_ty], [IntrConvergent]>;
```

Comment:
I see `int_dx_wave_is_first_lane` doesn't have `IntrNoMem` either but I know `IntrNoMem` was required to scalarize other intrinsics and this is an intrinsic we want to scalarize. Is the `IntrConvergent` attribute sufficient?

Reply:
My understanding was that `IntrNoMem` denotes there are no side-effects in the operation and that cross-lane communication would be a side-effect.